lib/dbutils.js

```
@@ -630,18 +656,27 @@ dbu.convertRows = function convertRows (rows, schema) {
  * CQL building for conditional requests in general.
  * @param {object} predicates, the 'attributes' object in queries.
  * @param {object} schema, the schema info for the logical table.
+ * @param {Boolean} noConvert if true, no attribute value conversion will take place
  * @return {object} queryInfo object with cql and params attributes
  */
-dbu.buildCondition = function buildCondition (predicates, schema) {
+dbu.buildCondition = function buildCondition (predicates, schema, noConvert) {
     function convert(key, val) {
         var convObj = schema.conversions[key];
-        if (convObj && convObj.write) {
+        if (!noConvert && convObj && convObj.write) {
             return convObj.write(val);
         } else {
             return val;
         }
     }
 
+    // make sure we have got a predicate object
+    if(!predicates || predicates.constructor !== Object) {
+        return {
+            cql: predicates || '',
```

I think that is actually a bug. The string case should only pertain to 'if not exists', but not to the query part where we perform a CAS by comparing with existing values, specified in an object. We should throw an error if query.if is a string, but does not normalize to 'if not exists'.